src/phasorpy/components.py

```diff
@@ -123,6 +132,208 @@ def two_fractions_from_phasor(
+    real: ArrayLike,
+    imag: ArrayLike,
+    components_real: ArrayLike | None = None,
+    components_imag: ArrayLike | None = None,
+    /,
+    *,
+    components_matrix: NDArray[Any] | None = None,
+    **kwargs: Any,
+) -> tuple[NDArray[Any], ...]:
+    """Return fractions of multiple components from phasor coordinates.
+
+    Parameters
+    ----------
+    real : array_like
+        Real component of phasor coordinates.
+        The harmonics must be in the first dimension.
+    imag : array_like
+        Imaginary component of phasor coordinates.
+        The harmonics must be in the first dimension.
+    components_real: array_like, optional
+        Real coordinates of the components.
+        The harmonics must be in the first dimension.
+    components_imag: array_like, optional
+        Imaginary coordinates of the components.
+        The harmonics must be in the first dimension.
+    components_matrix : array_like, optional
+        Components coefficient matrix.
+        The shape of the matrix must be (2h+1, n), where `h` is
+        the number of harmonics and `n` is the number of components.
+        Can be generated from components coordinates using
+        :py:func:`components_matrix`.
+    **kwargs : optional
+        Additional arguments passed to :py:func:`scipy.linalg.lstsq()`.
+
+    Returns
+    -------
+    fractions : tuple of ndarray
+        Components fractions.
+
+    Raises
+    ------
+    ValueError
+        The array shapes of `real` and `imag` do not match.
+        Neither the components matrix nor the components phasor
+        coordinates are not provided.
+        The system is undetermined due to the components matrix
+        having more columns than rows.
+        The number of harmonics in the components matrix does not
+        match the number of harmonics in the phasor coordinates.
+
+    See Also
+    --------
+    :ref:`sphx_glr_tutorials_api_phasorpy_components.py`
+
+    References
+    ----------
+    .. [2] Vallmitjana A, Lepanto P, Irigoin F, Malacrida L.
+      `Phasor-based multi-harmonic unmixing for in-vivohyperspectral
+      imaging <https://doi.org/10.1088/2050-6120/ac9ae9>`_.
+      *Methods Appl. Fluoresc.*, (2022)
+
+    Example
+    -------
+    >>> n_fractions_from_phasor(
+    ...     [0.5, 0.3],
+    ...     [0.2, 0.7],
+    ...     [0.1, 0.3],
+    ...     [0.2, 0.8],
+    ... )  # doctest: +NUMBER
+    (array([0.8162, 0.1513]), array([0.1958, 0.8497]))
+
+    """
+    real = numpy.atleast_1d(real)
+    imag = numpy.atleast_1d(imag)
+
+    if real.shape != imag.shape:
+        raise ValueError(f'{real.shape=} != {imag.shape=}')
+
+    if components_matrix is None:
+        if components_real is None or components_imag is None:
+            raise ValueError(
+                'either the components matrix or the components phasor '
+                'coordinates must be provided'
+            )
+        components_matrix = components_matrix(components_real, components_imag)
```

Comment:
The names of variable and function conflict. How about naming the function `phasor_component_matrix` so it can later be added to a flat phasorpy namespace?